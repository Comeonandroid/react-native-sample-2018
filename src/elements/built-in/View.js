import React from 'react'
import styled from 'styled-components'
import { PropTypes } from 'prop-types'


const View = ({ children, ...rest }) => (
  <Element {...rest}>
    {children}
  </Element>
)

View.propTypes = {
  children: PropTypes.node
}

const Element = styled.View `
`

export default View
