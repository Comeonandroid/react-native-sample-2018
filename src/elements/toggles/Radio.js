import React, { PureComponent } from 'react'
import { MKRadioButton } from 'react-native-material-kit'
import styled, { withTheme } from 'styled-components'
import { PropTypes } from 'prop-types'
import defaultsDeep from 'lodash/defaultsDeep'


import { View, Text } from '../index'

const defaultTextStyle = (theme) => ({
  color: theme.colors.font,
  fontWeight: '600',
  marginLeft: 4
})

class Radio extends PureComponent {

  constructor(params) {
    super(params)

    this.handleChange = this.handleChange.bind(this)
    this.handlePress = this.handlePress.bind(this)
  }

  handlePress() {
    this.handleChange({ checked: true })
  }

  handleChange({ checked }) {
    const { onChecked, onChange, name, value } = this.props
    if (checked && onChecked){
      onChecked(name, value)
    }
    if (onChange) { onChange(name, value) }
  }

  render() {
    const { children, checked, group, theme, textStyle, className, ...rest } = this.props
    return (
      <Wrapper>
        <Input
          checked={checked}
          group={group}
          fillColor={theme.colors.toggleFill}
          borderOnColor={theme.colors.font}
          borderOffColor={theme.colors.font}
          rippleColor={`rgba(250,166,50,.15)`}
          onCheckedChange={this.handleChange}
          className={className}
          {...rest}
        />
        <Text
          style={defaultsDeep(textStyle, defaultTextStyle(theme))}
          onPress={this.handlePress}>{children}</Text>
      </Wrapper>
    )
  }
}

const Wrapper = styled(View) `
  flex-direction: row;
  align-items: center;
  margin-top: 10px;
`

const Input = styled(MKRadioButton) `
  borderWidth: 3;
`

Radio.propTypes = {
  children: PropTypes.node,
  checked: PropTypes.bool,
  name: PropTypes.string,
  value: PropTypes.string,
  onChecked: PropTypes.func,
  onChange: PropTypes.func,
  group: PropTypes.object,
  textStyle: PropTypes.object,
  className: PropTypes.string,
  theme: PropTypes.object
}

Radio.defaultProps = {
  checked: false
}

export default withTheme(Radio)