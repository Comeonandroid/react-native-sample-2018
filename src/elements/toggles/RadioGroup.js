import React, { PureComponent } from 'react'
import { MKRadioButton } from 'react-native-material-kit'
import styled from 'styled-components'
import { PropTypes } from 'prop-types'

const group = new MKRadioButton.Group()

class RadioGroup extends PureComponent {
  render() {
    return (
      <Wrapper>
        {this.props.children(group)}
      </Wrapper>
    )
  }
}

const Wrapper = styled.View `
`

RadioGroup.propTypes = {
  children: PropTypes.func
}

export default RadioGroup