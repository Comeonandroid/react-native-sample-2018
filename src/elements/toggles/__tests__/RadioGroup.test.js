import 'react-native'
import React from 'react'
import { shallow } from 'enzyme'

import View from '../../built-in/View'
import RadioGroup from '../RadioGroup'
import Radio from '../Radio'

describe('<RadioGroup />', () => {
  it('renders correctly', () => {
    const wrapper = shallow(
      <RadioGroup>
        {(group) => (
          <View>
            <Radio group={group}>CNY</Radio>
            <Radio group={group}>USD</Radio>
            <Radio group={group}>ETH</Radio>
          </View>
        )}
      </RadioGroup>
    )
    expect(wrapper).toMatchSnapshot()
  })
})
