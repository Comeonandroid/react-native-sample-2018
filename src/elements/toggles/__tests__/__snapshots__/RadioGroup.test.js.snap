// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RadioGroup /> renders correctly 1`] = `
<Styled(Component)>
  <View>
    <WithTheme(Radio)
      group={
        Group {
          "buttons": Array [],
          "onAdd": undefined,
          "onRemove": undefined,
        }
      }
    >
      CNY
    </WithTheme(Radio)>
    <WithTheme(Radio)
      group={
        Group {
          "buttons": Array [],
          "onAdd": undefined,
          "onRemove": undefined,
        }
      }
    >
      USD
    </WithTheme(Radio)>
    <WithTheme(Radio)
      group={
        Group {
          "buttons": Array [],
          "onAdd": undefined,
          "onRemove": undefined,
        }
      }
    >
      ETH
    </WithTheme(Radio)>
  </View>
</Styled(Component)>
`;
